### PR TITLE
Chore/baip 1642

### PR DESCRIPTION
**Pull request by reno1979**
dependency update 

The styled components can not be updated any further, version >= 6 will not work with ASC (ASC uses 5 )  and version 6 does seem to have multiple issues

I am not sure about the peer dependencies

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1,2 +1,2 @@
-v16.5.0
+v18.15.0
```
Comment (fortunato) [inline .nvmrc:1]: Was v20 a leg too far? It becomes the active LTS from 2023-10-24 already.

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+legacy-peer-deps=true
```
Comment (fortunato) [inline .npmrc:1]: Ah didn't know we could do this! Life-saver :sweat_smile:

File: .eslintrc.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ module.exports = {
 	},
 	extends: [
 		'eslint:recommended',
-		'plugin:react/recommended',
+		'plugin:react/jsx-runtime',
```
Comment (fortunato) [inline .eslintrc.js:8]: Can I ask about he rationale behind this?

File: .stylelintrc.js [hunk 4]
```diff
@@ -1,114 +1,125 @@
...
-		'value-list-comma-newline-after': 'always-multi-line',
-		'value-list-comma-space-after': 'always-single-line',
-		'value-list-comma-space-before': 'never',
-		'value-list-max-empty-lines': 0,
-		'value-no-vendor-prefix': true,
-		'string-quotes': 'single',
+		'selector-class-pattern': null,
+		'media-query-no-invalid': null,
 	},
+	// overrides: [
```
Comment (fortunato) [inline .stylelintrc.js:9]: Can this be removed?

File: jest.config.js [hunk 5]
```diff
@@ -168,7 +163,7 @@ module.exports = {
 	//     // '^.+\\.(ts|tsx)?$': `<rootDir>/tests/jest-preprocess.js`
 	// },
 	transform: {
-		'^.+\\.[t|j]sx?$': 'babel-jest',
+		'^.+\\.[t|j]sx?$': '@swc/jest',
```
Comment (fortunato) [inline jest.config.js:166]: Legend :slightly_smiling_face:

File: .babelrc.js [hunk 6]
```diff
@@ -23,7 +23,12 @@ module.exports = (api) => {
 					corejs: 3,
 				},
 			],
-			'@babel/preset-react',
+			[
```
Comment (fortunato) [inline .babelrc.js:26]: Why not build with SWC as well and get rid of this?

File: .nvmrc (hunk 1 above)
Comment (reno1979) [reply .nvmrc:1]: cool

File: .babelrc.js (hunk 6 above)
Comment (reno1979) [reply .babelrc.js:26]: wel we had some serious issues with Jest but I will change it to see what happens

File: .eslintrc.js (hunk 3 above)
Comment (reno1979) [reply .eslintrc.js:8]: The import React from 'react'; is removed, therefor we have to tell the linter about this (we are using the new jsx transform) See comment : 

```If you are using the [new JSX transform from React 17](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports), extend [react/jsx-runtime](https://github.com/jsx-eslint/eslint-plugin-react/blob/c8917b0885094b5e4cc2a6f613f7fb6f16fe932e/index.js#L163-L176) in your eslint config (add "plugin:react/jsx-runtime" to "extends") to disable the relevant rules.```

source: https://github.com/jsx-eslint/eslint-plugin-react

Comment (fortunato) [reply .eslintrc.js:8]: Cool, thanks for clarifying :pray:

**Comment by bennyvdhoogen** [comment]
Just going to mention this since the `package.lock` has so many changes: is this tested with the production/minified js?

**Pull request merged**